### PR TITLE
Miscellaneous warning fixes

### PR DESCRIPTION
**Pull request by thesamesam**

### CODE REVIEW

File: src/libeinfo/libeinfo.c [hunk 1]
```diff
@@ -677,7 +677,7 @@ ewarn(const char *EINFO_RESTRICT fmt, ...)
 void
 ewarnx(const char *EINFO_RESTRICT fmt, ...)
 {
-	int retval;
+	_unused int retval;
```
Comment (vapier) [inline src/libeinfo/libeinfo.c:680]: the warning looks correct.  why don't we delete it.

File: src/shared/_usage.h [hunk 2]
```diff
@@ -44,6 +44,12 @@
 	case 'q': case_RC_COMMON_getopt_case_q; break;			      \
 	default:  case_RC_COMMON_getopt_default; break;
 
+#if __GNUC__ > 2 || defined(__INTEL_COMPILER)
```
Comment (vapier) [inline src/shared/_usage.h:47]: this looks like it belongs in `helpers.h` instead

File: src/librc/librc.c [hunk 3]
```diff
@@ -755,8 +755,10 @@ rc_service_mark(const char *service, const RC_SERVICE state)
 					    RC_SVCDIR "/%s/%s",
 					    rc_parse_service_state(RC_SERVICE_WASINACTIVE),
 					    base);
-					if (symlink(init, was) == -1)
+					if (symlink(init, was) == -1) {
+						free(init);
```
Comment (vapier) [inline src/librc/librc.c:759]: this looks correct, but from scanning the rest of this func, seems like there's other places this leaks that need fixing too ?

**Review by vapier — CHANGES_REQUESTED** [review]
i merged some, so you'll want to rebase

File: src/librc/librc.c (hunk 3 above)
Comment (thesamesam) [reply src/librc/librc.c:759]: sorry, I can't spot it - could you point one out?

Comment (vapier) [reply src/librc/librc.c:759]: sorry, i was looking at the file w/out your patch applied, and thought it was fixing an earlier line

File: src/openrc-shutdown/broadcast.c [hunk 4]
```diff
@@ -33,6 +33,7 @@
 
 #include "broadcast.h"
 #include "helpers.h"
+#include "_usage.h"
```
Comment (vapier) [inline src/openrc-shutdown/broadcast.c:36]: don't need this include anymore

**Review by vapier — CHANGES_REQUESTED** [review]
merged most things.  the 2 patches left i think you can/should squash into one.

File: src/librc/librc.c (hunk 3 above)
Comment (thesamesam) [reply src/librc/librc.c:759]: np

File: src/openrc-shutdown/broadcast.c (hunk 4 above)
Comment (thesamesam) [reply src/openrc-shutdown/broadcast.c:36]: ugh, I thought I'd got that - thanks

**Comment by vapier** [comment]
thx bud

**Comment by thesamesam** [comment]
cheers!

**Pull request merged**